Enterprise-CMCS/macpro-mdct-mcr
# Remove adminDisabled from report forms

### Description
**Context:** Admins are not allowed to edit reports, but there are certain forms they _can_ edit, such as the admin banner. All forms on the site are implemented on the same foundation, Form.tsx. Therefore it has to account for both.

**Current state:** We disallow admins from editing reports by marking report forms as `adminDisabled`. Since a report can contain many forms, sometimes in nested routes, we have a function `copyAdminDisabledStatusToForms` which will ensure all forms in a report have the flag set. This is a proper pain in the butt when comparing form templates.

**This PR's change:** We will invert this flag to `editableByAdmins`, and put it on the special admin forms instead. This allows us to delete all of the special handling from the report form templates.

<sup>Note: I changed mcpar.json and mlr.json in app-api, but not the copies in ui-src, because those copies will soon disappear thanks to ~11406~ #11408. _Edit: Done!_</sup>

### Related ticket(s)
n/a

---
### How to test
* Log in as an admin. Ensure you can edit the admin banner, but not reports.
* Log in as a regular user. Ensure you can create a report and enter data.

### Important updates
n/a

---
### Author checklist
